README.md

```diff
@@ -1,18 +1,84 @@
+to implement playback experience, and if you have different logic, you just override that particular class or functionality without the need to thoroughly understand the whole code base. As the result, you are able to easily and quickly use TubiPlayer without invest much time 
+upfront.
+
+Nearly every business related dependencies are managed by [PlayerModuleDefault](./lib/src/main/java/com/tubitv/media/di/PlayerModuleDefault.java), where all the dependencies are being instantiated and being injected into [DoubleViewTubiPlayerActivity](./lib/src/main/java/com/tubitv/media/activities/DoubleViewTubiPlayerActivity.java)
+You can choose to use the default behavior by using the module out of box, but if you have customized business logic, you can directly change the dependencies in ***PlayerModuleDefault*** for customization
+
+## How to use
+Different application often requires different sets of rules and logic to satisfy business requirements, as the result, Tubiplayer was build to provide high degree of
+customization. However, highly customized code if not managed well can quickly turn into nightmare. Therefore, Tubiplayer leverages a third party Dependency Injection framework:[Dagger](https://github.com/google/dagger) 
+to provide the best of both worlds.(If you are not familiar with using dagger, please take sometime to learn it, because it is a very good tool to reduce complexity, testing, and other software development merits)
+
+### Different use cases:
+1. If you just simply want to play a video without any interruption, you can just use ***DoubleViewTubiPlayerActivity*** out of box, then the code start a self contained activity to handle your playback experience
+```java
+      String subs = "http://put_your_own_subtitle.srt";
+                 String artwork = "http://www.put_your_own_art_work.png";
+                 String name = "Example Video";
+                 String video_url = "http://put_your_own_video_url.mp4";
+                 Intent intent = new Intent(SelectionActivity.this, DoubleViewTubiPlayerActivity.class);
+                 intent.putExtra(TubiPlayerActivity.TUBI_MEDIA_KEY, MediaModel.video(name, video_url, artwork, null));
+                 startActivity(intent);
+```
+
+2. The playback is still handle by standalone **DoubleViewTubiPlayerActivity** by using the above usage. In addition to play a video, if you also want to implement pre-roll, and middle-roll video ads, then there are more steps needed:
+    1. First the presumption is all the video ads need to be real time and dynamic, meaning all the ads are not fetched until they are needed.
+    2. Therefore, You need to know at beginning of playback at which positions has opportunity to show ads, in another words the list of positions during the main video to fetch ads from server, we called it the ***CuePoints***.
```

Comment:
Erase bullet point one, let's combine 1 and 2 together:
1. You need to know at beginning of playback at which positions ads may need to show. All the video ads need to be real time and dynamic, meaning they are not fetched until they are needed. 
 We call the list of positions during the main video to fetch ads from server ***CuePoints***.